prov/tcp/src/xnet.h

```diff
@@ -583,12 +583,12 @@ xnet_set_commit_flags(struct xnet_xfer_entry *xfer, uint64_t flags)
 }
 
 static inline uint64_t
-xnet_tx_completion_flag(struct xnet_ep *ep, uint64_t op_flags)
+xnet_tx_completion_get_msgflags(struct xnet_ep *ep, uint64_t flags)
 {
-	/* Generate a completion if op flags indicate or we generate
-	 * completions by default
+	/* Generate a completion if msg flags indicate or app
+	 * requests
 	 */
-	return (ep->util_ep.tx_op_flags | op_flags) & FI_COMPLETION;
+	return (ep->util_ep.tx_msg_flags | flags) & FI_COMPLETION;
```

Comment:
`tx_msg_flags` and `tx_op_flags` are the same w.r.t. `FI_COMPLETION`. There is no behavior difference before and after this patch.  The `*_msg_flags` fields are used as a way to pass optional `FI_COMPLETION` to other util calls (such as in `util_srx_recvmsg()`). The `*_op_flags` should be sufficient to cover all the scenarios that the default flags are checked.

Reply:
@shefty You are right. When enabling completion by default while using selective completions, the msg flags won't have FI_COMPLETION on.